packages/flutter_tools/lib/src/base/os.dart

```diff
@@ -103,6 +103,9 @@ abstract class OperatingSystemUtils {
   /// Return the File representing a new pipe.
   File makePipe(String path);
 
+  /// Return a directory's total size in human-readable format.
+  String? directorySize(Directory directory);
```

Comment:
Consider renaming this to something like `directorySizeAsMB` as currently this always returns megabytes. We can always rename this later if we support more units :)

```suggestion
  String? directorySizeAsMB(Directory directory);
```

Reply:
Agreed that seems reasonable; or break up into two functions:
* One (maybe `getDirectorySize`) that takes a `Directory` and returns the directory size in bytes. 
* One (maybe `formatSizeAsMB`) that takes an int number of bytes and returns a human-readable string (which could be placed in some utils file maybe).

Reply:
Renamed 👍